src/gerbil/runtime/mop.ss

```diff
@@ -6,449 +6,554 @@ package: gerbil/runtime
 namespace: #f
 
 (export #t)
-(import "gambit" "util")
+(import "gambit" "util" "c3")
 
-;; Gerbil rtd [runtime type descriptor]:
-;;  {##struct-t id super fields name plist ctor slots methods}
-;;  {##class-t  id super fields name plist ctor slots methods}
-;;
-;; Gambit structure rtd:
+;; Gambit structure rtd [runtime type descriptor]
 ;;  (define-type type
 ;;    (id      unprintable: equality-test:)
 ;;    (name    unprintable: equality-skip:)
 ;;    (flags   unprintable: equality-skip:)
 ;;    (super   unprintable: equality-skip:)
 ;;    (fields  unprintable: equality-skip:))
 ;;
-;; Gerbil rtd on gambit
-;; ##structure ##type-type
-;;  1  ##type-id
-;;  2  ##type-name
-;;  3  ##type-flags
-;;  4  ##type-super
-;;  5  ##type-fields
-;;  6                       type-descriptor-mixin
-;;  7                       type-descriptor-fields
-;;  8                       type-descriptor-plist
-;;  9                       type-descriptor-ctor
-;; 10                       type-descriptor-slots
-;; 11                       type-descriptor-methods
+;; Gerbil rtd on Gambit, extending the above (index, accessor, type)
+;;  0  ##structure-type                 : (Exactly ##type-type)
+;;  1  ##type-id                        : Symbol ; sometimes uninterned
+;;  2  ##type-name                      : Symbol
+;;  3  ##type-flags                     : Fixnum
+;;  4  ##type-super                     : (OrFalse StructTypeDescriptor)
+;;  5  ##type-fields                    : (Vector [Symbol Fixnum default-value] ...)
+;;  6  type-descriptor-precedence-list  : (List TypeDescriptor) ; doesn't contain the class itself
+;;  7  type-descriptor-all-slots        : (Vector (OrFalse Symbol)) ; first is false
```

Comment:
we could call the first one `class` as it is the type descriptor itself I guess. This would give us a nice `.class` notation for taking the type of an object moving forward, what do you think?